lib/project/pro_motion/activities/pm_navigation_activity.rb

```diff
@@ -33,31 +33,31 @@ def close_fragment
     end
 
     def fragment
-      self.fragments.last
-    end
-
-    def fragments
-      @fragments ||= []
+      # self.fragments.last
+      getFragmentManager.findFragmentByTag("screen-#{fragmentManager.getBackStackEntryCount}")
     end
 
     def on_fragment_attached(frag)
-      self.fragments << frag
+      # No-op for now.
     end
 
     def on_fragment_detached(frag)
-      self.fragments.pop
+      # No-op for now.
     end
 
     private
 
     def activity_init
       setup_root_fragment_container
-      setup_root_fragment root_fragment_instance
-      open_fragment root_fragment_instance if root_fragment_instance
+      @root_fragment ||= intent_fragment_class && Kernel.const_get(intent_fragment_class.to_s).new
+      return unless @root_fragment
+      setup_root_fragment @root_fragment
+      open_fragment @root_fragment
+      @root_fragment = nil # Don't hang onto this reference.
```

Comment:
We also don't want to hang onto a @root_fragment reference either. So I'm clearing it here, and I brought the `root_fragment_instance` code into this method since it didn't make sense to keep it separate now.